### PR TITLE
Feature: list structure

### PR DESCRIPTION
**Pull request by memelopez**
This PR implements:
- new project with webpack configuration
- index.html file that is a template using HTML webpack plugin
- hardcoded tasks array with description, completed, and index
- functions that iterates over tasks array and populates HTML list
- on page load render dynamically created list
- styles.css that works with CSS loader

### CODE REVIEW

File: src/ui.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    const iconRemove = document.createElement('I'); // creates remove icon
+    iconRemove.className = 'fas fa-trash p-2 d-none removeIcn';
+    div4Icons.appendChild(iconRemove); // appends remove icon to item
+
+    item.appendChild(divNormal);
+    // item.appendChild(divEdit);
+    item.appendChild(div4Icons);
+
+    list.appendChild(item); // appends item to list
+  }
```
Comment (EricMbouwe) [inline src/ui.js]: - [ ] There is an issue when the index of the todos is changed, the change doesn't reflect in the UI. This is due to the fact that you are not taking care of the todo index in your implementation of these two functions. Let's fix that and have synchronization between the state and the UI.

- [x] Let's follow the HTML best practices by using lowercase for elements and attribute names.
```
document.createElement('LI')
document.createElement('DIV')
document.createElement('I')
document.createElement('INPUT')
etc.
```
These are bad practices.

Let's use these ones
```
document.createElement('li')
document.createElement('div')
document.createElement('i')
document.createElement('input')
```

File: src/ui.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  static addbottombtn() {
+    const appDiv = document.querySelector('#appDiv');
+
+    const btmDiv = document.createElement('DIV');
+    btmDiv.className = 'd-flex justify-content-center align-items-center border btmDiv';
+
+    const pBtm = document.createElement('P');
```
Comment (EricMbouwe) [inline src/ui.js]: - [x]  Let's follow the HTML best practices by using lowercase for elements and attribute names. Check the comment above for more details.

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @memelopez,

Good job so far!
- [x] Good descriptive PR 👍
- [x] Good Readme 👍
- [x] Good commits 👍
 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/ui.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    // item.appendChild(divEdit);
+    item.appendChild(div4Icons);
+
+    list.appendChild(item); // appends item to list
+  }
+
+  static addbottombtn() {
+    const appDiv = document.querySelector('#appDiv');
+
+    const btmDiv = document.createElement('DIV');
```
Comment (EricMbouwe) [inline src/ui.js]: - [x] You missed this one! Let's change it and you should be good to go to the next step.

File: src/ui.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  }
+
+  static addTasksUI(tasks) {
+    // Iterates over array tasks to populate HTML list
+    if (tasks.length === 0) {
+      this.addEmptyToDoMessage();
+    } else {
+      tasks.forEach((task) => this.addTaskToList(task));
+    }
+  }
```
Comment (EricMbouwe) [inline src/ui.js]: - [x] The dynamic rendering is not yet correct. If you change the index of the first task in your task array to `1` and the second one to `0` you still have the same render as if not changed. As the requirements say `On page load, the task array is rendered dynamically to the DOM. The tasks should appear in order of the index values for each task.` Please fix that!

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @memelopez,

Great work on making the changes requested by a previous reviewer 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Elmer, 

Your project is complete! Great job for making the necessary changes recommended by the previous reviewer(s). There is nothing else to say other than... it's time to merge it 🍾:shipit:.
Congratulations! 🎉🥂

_Kindly consider my comment(s) marked **[OPTIONAL]** under this review:_
<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have no more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**